src/core/ddsi/include/dds/ddsi/shm_transport.h

```diff
@@ -71,6 +72,12 @@ DDS_EXPORT iceoryx_header_t *iceoryx_header_from_chunk(const void *iox_chunk);
 
 DDS_EXPORT void shm_set_loglevel(enum ddsi_shm_loglevel);
 
+DDS_EXPORT void *shm_create_chunk(iox_pub_t iox_pub, size_t size);
```

Comment:
I've tried to make sure all files that one can "reasonably expect" to be used by applications needing some special features consistently prefix all names with `dds_` or `ddsi_` (or uppercase; all depending on the circumstances; just compare `q_config.h` @ aba5e78760b9415d21beb8319b0af0c2b730ce9a with `ddsi_config.h` now). I know this `iox_`-, 'iceoryx_`- and `shm_`-prefixed stuff has been around for a while, but some of this slowly morphed from internal-only (where I don't care) to somewhat-public-API. Continuing down this path seems unwise, and indeed, I think the pre-existing symbols ought to be cleaned up.

I should've paid more attentive to that ...

Rename all that in the same PR as adding this new functionality doesn't seem like a great idea. That leaves us with: first fix all the existing names, then do this PR with updated names; or do this PR fitting the pattern, then do a PR updating all the names. Practical considerations make me willing to accept the latter.

Reply:
I prefer the latter option and can prepare a renaming PR after this one, but as you say I would prefer not to mix it up with functionality changes.